app/src/main/kotlin/com/flxrs/dankchat/main/MainFragment.kt

```diff
@@ -403,12 +407,30 @@ class MainFragment : Fragment() {
                         isUserChange = false,
                         loadTwitchData = true,
                     )
-                    val name = dankChatPreferences.userName
-                    if (dankChatPreferences.isLoggedIn && name != null) {
-                        showSnackBar(getString(R.string.snackbar_login, name))
+                    // wait for username to be validated before showing snackbar
+                    collectFlow(dankChatViewModel.currentUserName) {
+                        if (dankChatPreferences.isLoggedIn && it != null) {
+                            showSnackBar(getString(R.string.snackbar_login, it))
+                        }
                     }
                 }
             }
+            // handle oAuth validation error
+            collectFlow(dankChatViewModel.validationError) {
+                when (it) {
+                    OAuthValidationError.OAuthTokenInvalid      -> {
+                        // I'm too donk to create a Fragment or something sorry
+                        MaterialAlertDialogBuilder(requireContext())
+                            .setTitle(getString(R.string.oauth_expired_title))
+                            .setMessage(getString(R.string.oauth_expired_message))
+                            .setPositiveButton(getString(R.string.oauth_expired_login_again)) { _, _ -> navigateSafe(R.id.action_mainFragment_to_loginFragment) }
+                            .setNegativeButton(getString(R.string.oauth_expired_dismiss)) { _, _ -> } // default action is dismissing anyway
```

Comment:
Is `Dismiss` or `Later` a negative button or should we maybe make it a neutral one? Main difference is the button placement in the dialog

Reply:
I think the `Dismiss` or `Later` button is negative in sense of "Do you want to login again?" -> Yes = Login again, No = Dismiss.